copybookreader-plugins/docs/CopyBookReader-batchsource.md

```diff
@@ -0,0 +1,99 @@
+# CopybookReader plugin
+
+
+Description
+-----------
+This is a source plugin that would allow users to read and process mainframe files defined using COBOL Copybook.
+This is the basic first implementation.
+
+Use Case
+--------
+
+It's basically used for reading flat file or dataset that is generated on a z/OS IBM mainframe based on a fixed length
+COBOL Copybook. This will also work on AS/400 computers.
+So, if a customer has flat files on HDFS that can be parsed using simple COBOL Copybook, then applying the Copybook
+one is able to read the file and it's fields easily.
+
+
+Properties
+----------
+
+**CopybookContents :** Contents of the COBOL Copybook file which will contain the data structure
+User will have to paste the entire contents of the Copybook file in the test area.
+The Copybook contents should be in sync with the underlying binary data to be read.
+First implementation does not handle complex nested structures of COBOL Copybook
+Also it does not handle Redefines or iterators in the structure
+
+**binaryFilePath   :** Complete path of the .bin to be read.This will be a fixed length binary format file,that
+matches the Copybook.
+Supports compressed files - Native Compressed Codec
```

Comment:
```
**binaryFilePath:** Complete path of the .bin to be read. This will be a
fixed-length binary format file that matches the copybook. Supports
compressed files using a native compressed codec.